### PR TITLE
CLI configuration & interactive setup for Firewall

### PR DESCRIPTION
**Pull request by abhisharma404**
## Status
**READY**

## Description
**Summary** 
1. User can configure Firewall from CLI.
2. Interactive setup-mode for Firewall.

**Changes in `@takeInput`:**
  - Recursive algorithm to iterate over nested dictionaries: Earlier, we did not have nested configuration, however SecureTea-Firewall uses a highly nested configuration, `@takeInput` can now handle all nested dictionaries. 
  - Default fall-back configuration added: If user provides an empty value for a particular input, default fall-back is used, **highly necessary for Firewall**.
  - Future developers need not to worry about complexity of setting up an interactive mode for any feature they add, `@takeInput` is highly efficient to handle most cases.

**Changes in `arguments.py`:**
  - Argument option for each firewall configuration has been added. User can configure firewall from their CLI.

**Changes in `core.py`:**
  - In order to adapt to the above changes, it's been updated accordingly.

## Related PRs
None

## Todos
- [ ] Tests
- [x] Documentation
- [x] I have fully tested the changes locally

## Deploy Notes
None

## Steps to Test or Reproduce
`sudo python3 SecureTea.py --firewall`

## Impacted Areas in Application
`args_helper.py`
`arguments.py`
`core.py`

### CODE REVIEW

File: securetea/args/args_helper.py [hunk 1]
```diff
@@ -76,28 +145,14 @@ def configureApp():
...
-                        config_dict[key] = str(input('>> Enter {}: '
-                                                     .format(item))).strip()
-                        if (config_dict[key] == 's' or
-                            config_dict[key] == 'S'):
-                            skip = True
-            if not skip:
-                return config_dict
-            else:
-                return None
+            dict_value = func(*args)
```
[the remark this answers is not in the document]
Comment (abhisharma404) [reply securetea/args/args_helper.py]: @rejahrehim Again, Codacy is treating `func` passed to the decorator function as an object, not as  a function!

**Comment by adeyosemanputra** [comment]
Could u give me the screenshot of the result 
It's better u can write the documentation too (how to) 
https://github.com/OWASP/SecureTea-Project/wiki

**Comment by adeyosemanputra** [comment]
@abhisharma404 
Steps to Test or Reproduce
sudo python3 SecureTea.py --firewall
I thought it's better if till
sudo python SecureTea.py --firewall

**Comment by abhisharma404** [comment]
> @abhisharma404
> Steps to Test or Reproduce
> sudo python3 SecureTea.py --firewall
> I thought it's better if till
> sudo python SecureTea.py --firewall
> Or SecureTea.py --firewall

@adeyosemanputra No worries! :) The code is equally compatible with Python 2 & 3, I have tested on both of them.
Talking about `sudo`, since Firewall uses Scapy and NetfilterQueue to bind with the iptables, we need a root privilege.

As requested, I have added GIF screencasts running the commands:
 - **Command:** `sudo python SecureTea.py`
Firewall2

 -  **Command:** `sudo python3 SecureTea.py`
Firewall1

Also, I'm not able to edit the WiKi, could you please setup, I'll send a PR regarding the documentation then.

I hope it helps, please suggest any changes if needed, I'll be glad to do so. :)
Regards

**Pull request merged**